sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -151,6 +156,35 @@ public static SqlTransform query(String queryString) {
         .build();
   }
 
+  /** Load all UDF/UDAF from {@link UdfUdafRegister}. */
+  public SqlTransform autoRegisterUdfUdaf() {
+    List<UdfDefinition> newUdfs = new ArrayList<>();
+    List<UdafDefinition> newUdafs = new ArrayList<>();
+
+    for (UdfUdafRegister ins :
+        Lists.newArrayList(ServiceLoader.<UdfUdafRegister>load(UdfUdafRegister.class))) {
```

Comment:
`ServiceLoader` implements `Iterable` interface so you don't need to wrap it into a arraylist to do `for each`.